stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+            "description": "Instance type for workers.",
+            "default": "t2.small",
+        },
+        "KeyName": {
+            "type": str,
+            "description": "Existing SSH key name for worker access."
+        },
+        "RootVolumeSize": {
+            "type": int,
+            "description": "Root volume size in GB.",
+            "default": 20,
+        },
+        "RootVolumeDevice": {
+            "type": str,
+            "description": "The block device name for the root volume. This "
+                           "will depend on instance type and AMI",
+            "default": "/dev/sda1"
+        }
+    }
+
+    def create_node_instance_role(self):
+        t = self.template
+
+        # The user data below relies on this map being present.
+        create_max_pods_per_node_mapping(t)
+
+        # This re-creates NodeInstanceRole from Amazon's CFN template
+        role = t.add_resource(
+            Role(
+                "NodeInstanceRole",
+                AssumeRolePolicyDocument=Policy(
+                    Statement=[
+                        Statement(
+                            Effect=Allow,
+                            Action=[AssumeRole],
+                            Principal=Principal("Service", ["ec2.amazonaws.com"])
+                        )
+                    ]
+                ),
+                Path="/",
+                ManagedPolicyArns=[
+                    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+                    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+                    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+                ]
+            )
+        )
+        t.add_output(Output("NodeInstanceRole", Value=role.ref()))
+        return role
+
+    def create_template(self):
+        t = self.template
+        variables = self.get_variables()
+
+        # Create the node instance profile which allows nodes to join the
+        # EKS Cluster
+        role = self.create_node_instance_role()
+        profile = t.add_resource(
+            InstanceProfile(
+                "NodeInstanceProfile",
+                Roles=[role.ref()]
+            )
+        )
+
+        cluster_name = variables["ClusterName"]
+        instance_type = variables["InstanceType"]
+        user_data = get_launch_config_userdata(cluster_name, instance_type)
+
+        # Create the launch configuration with a userdata payload that
+        # configures each node to connect to
+        launch_config = t.add_resource(
+            LaunchConfiguration(
+                "NodeLaunchConfig",
+                AssociatePublicIpAddress=False,
+                IamInstanceProfile=profile.ref(),
+                ImageId=variables["ImageId"],
+                InstanceType=variables["InstanceType"],
+                KeyName=variables["KeyName"],
+                SecurityGroups=[variables["WorkerSecurityGroupId"]],
+                UserData=Base64(Join("", user_data)),
```

Comment:
The method `get_launch_config_userdata` already returns a `Base64` and `Join`'d string, right?  Not sure this will work here.

Reply:
nice catch